powermon/libs/configurationManager.py

```diff
@@ -0,0 +1,150 @@
+            formatter = hass(formatConfig, device)
+        elif formatType == FormatterType.TOPICS:
+            from powermon.formats.topics import Topics
+            formatter = Topics(formatConfig, topic)
+        elif formatType == FormatterType.SIMPLE:
+            from powermon.formats.simple import simple
+            formatter = simple(formatConfig)
+
+        return formatter
+    
+    @staticmethod
+    def parseOutputConfig(outputConfig: dict, topic: str, schedule_name: str, device: Device, mqtt_broker: MqttBroker) -> AbstractOutput:
+        outputType = outputConfig["type"]
+
+        formatConfig = outputConfig["format"]
+        topic_override = outputConfig.get("topic_override", None)
+        #use topic override from the config
+        if(topic_override is not None):
+            topic = topic_override
+
+        format = ConfigurationManager.getFormatfromConfig(formatConfig, device, topic)
+
+        output_class = None
+        #Only import the required class
+        if outputType == OutputType.SCREEN:
+            from powermon.outputs.screen import Screen
+            output_class = Screen(outputConfig, format)
+        elif outputType == OutputType.MQTT:
+            from powermon.outputs.mqtt import MQTT
+            output_class = MQTT(outputConfig, topic, mqtt_broker, format)
+        elif outputType == OutputType.API_MQTT:
+            from powermon.outputs.api_mqtt import API_MQTT
+            output_class = API_MQTT(outputConfig, topic, schedule_name, mqtt_broker, format)
+            
+
+        return output_class
+    
+    @staticmethod
+    def parseCommandConfig(command_config : dict, topic_prefix : str, mqtt_broker : MqttBroker, device: Device):
+        
+        _command_query = command_config["command_query"]
+        _commandType = command_config["type"]
+        _schedule_name = command_config["schedule_name"]
+        results_topic = topic_prefix + _schedule_name + "/results/" + _command_query
+
+        _outputs = []
+        for outputConfig in command_config["outputs"]:
+            _output = ConfigurationManager.parseOutputConfig(outputConfig, results_topic, _schedule_name, device, mqtt_broker)
+            logging.debug(f"output type: {_output}")
+            _outputs.append(_output)
+
+        _command = None
+        if _commandType == CommandType.POLL:
+            _command = PollCommand(_command_query, _commandType, _schedule_name, _outputs, device.get_port())
+        
+        return _command
+    
+    @staticmethod
+    def parseControllerConfig(config : dict, device : Device, mqtt_broker : MqttBroker) -> ScheduleController:
+        logging.debug("parseCoordinatorConfig")
+        logging.debug(f"config: {config}")
+        _name = config["name"]
+        _loopDuration = config["loop_duration"]
+        if(_loopDuration == "once"):
+            _loopDuration = 0
+
+        _commands = []
+        topic_prefix = f"powermon/{_name}/results/"
```

Comment:
this should be a config item

Reply:
Yes, it needed to match with the API so I hardcoded it. I can include it in the announce message.